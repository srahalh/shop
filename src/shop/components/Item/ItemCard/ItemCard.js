import React from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import ItemImage from '../ItemImage/ItemImage';

import './ItemCard.scss';

const ItemCard = ({id, title, price, imageURL, category}) => {
  const navigate = useNavigate();
  return (
    <article
      className="ItemCard__container"
      onClick={() => navigate(`/product/${id}`)}
    >
      <ItemImage
        imageURL={imageURL}
      />
      <div className="ItemCard__info">
        <h3>{title}</h3>
        <p>Category: {category}</p>
        <p>Price:<strong> {price} â‚¬</strong></p>
      </div>
    </article>
  )
}

ItemCard.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  price: PropTypes.number, 
  imageURL: PropTypes.string, 
  category: PropTypes.string
}

export default ItemCard