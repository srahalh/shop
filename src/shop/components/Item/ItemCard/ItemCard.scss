@import '../../../styles/mixins.scss';
@import '../../../styles/config.scss';

.ItemCard__container {
  @include flex-layout-columns;
  align-items: initial;
  height: 21rem;
  padding: $padding-md;
  margin: $margin-sm $margin-sm;
  cursor: pointer;
  border: 1px solid lighten($color-primary, 70%);
  background-color: $color-background-2;

  @include box-shadow;

  &:hover {
    -webkit-transform: scale(1.05);
    transform: scale(1.05);
    transition: all 0.3s ease-out;
  }

  .ItemCard__info {
    margin-top: $padding-lg;
    border-top: 1px solid lighten($color-primary, 70%);
    h3 {
      width: 18rem;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }

  @include desktop {
    flex: 0 0 46.62%;
  }

  @include desktop-lg {
    flex: 0 0 22.96%;
  }
}