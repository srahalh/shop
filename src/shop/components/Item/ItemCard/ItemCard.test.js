import React from "react";
import { renderWithProviders } from "../../../test/setupTests";
import ItemCard from "./ItemCard";
const fulfilledState = require("../../../test/fulfilledProductDetailState.json");


describe("Test ItemCard", () => {
  test("Should render ItemCard", () => {
    const { asFragment  } = renderWithProviders(<ItemCard {...fulfilledState.products.queries["getProductById(\"9\")"].data[0]} />);
    expect(asFragment ()).toMatchSnapshot();
  });
});
