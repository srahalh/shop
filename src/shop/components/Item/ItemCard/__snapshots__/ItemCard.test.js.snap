// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemCard Should render ItemCard 1`] = `
<DocumentFragment>
  <article
    class="ItemCard__container"
  >
    <img
      alt="Some images here"
      class="ItemImage__container"
      loading="lazy"
    />
    <div
      class="ItemCard__info"
    >
      <h3>
        Liquid 5
      </h3>
      <p>
        Brand: ASUS
      </p>
      <p>
        Price:
        <strong>
           20 â‚¬
        </strong>
      </p>
    </div>
  </article>
</DocumentFragment>
`;
